tensorflow/core/distributed_runtime/remote_device.cc

```diff
@@ -67,12 +67,12 @@ void NewRemoteDevices(Env* env, WorkerCacheInterface* worker_cache,
     GetStatusResponse resp;
   };
   Call* call = new Call;
-  auto cb = [env, worker_cache, worker_name, done, wi,
+  auto cb = [env, worker_cache, &worker_name, &done, wi,
```

Comment:
Is this safe? `worker_name` is a `const string&` and `cb` might be called after `NewRemoteDevices()` returns and `worker_name` no longer refers to the same object. (Similarly, it seems as if `done` will be destroyed before the version captured by reference in this change is called in the callback.)